### PR TITLE
Standard and core library inline documentation

### LINKED ISSUE #4812: Update standard library and core library documentation to follow inline documentation standard
Inline documentation standard can be seen here: https://github.com/FuelLabs/sway-standards/tree/master/standards/src_2

Core library:
- [x] never.sw
- [x] ops.sw
- [x] primitive_conversions.sw
- [x] primitives.sw
- [x] raw_ptr.sw
- [x] raw_slice.sw
- [x] storage.sw

Standard library:
- [x] storage/storable_slice.sw
- [x] storage/storage_api.sw
- [x] storage/storage_bytes.sw
- [x] storage/storage_key.sw
- [x] storage/storage_map.sw
- [x] storage/storage_string.sw
- [x] storage/storage_vec.sw
- [x] vm/evm/ecr.sw
- [x] vm/evm/evm_adress.sw
- [x] address.sw
- [x] alloc.sw
- [x] assert.sw
- [x] auth.sw
- [x] b256.sw
- [x] b512.sw
- [x] block.sw
- [x] bytes.sw
- [x] call_frames.sw
- [x] constants.sw
- [x] context.sw
- [x] contract_id.sw
- [x] convert.sw
- [x] ecr.sw
- [x] error_signals.sw
- [x] external.sw
- [x] flags.sw
- [x] hash.sw
- [x] identity.sw
- [x] inputs.sw
- [x] instrinsics.sw
- [x] logging.sw
- [x] low_level_call.sw
- [x] math.sw
- [x] message.sw
- [x] option.sw
- [x] outputs.sw
- [x] registers.sw
- [x] result.sw
- [x] revert.sw
- [x] string.sw
- [x] token.sw
- [x] tx.sw
- [x] u128.sw
- [x] u256.sw
- [x] vec.sw

### PR DESCRIPTION
**Pull request by SwayStar123**
## Description
Standardized documentation for the standard library and the core library. Following the inline documentation standard (src-2)

Closes #4812

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lib-std/src/option.sw [hunk 1]
```diff
@@ -162,13 +182,21 @@ impl<T> Option<T> {
 
     /// Returns the contained `Some` value or a provided default.
     ///
-    /// `unwrap_or`: `Option::unwrap_or`
+    /// # Arguments
     ///
-    /// ### Examples
+    /// * `defulat`: [T] - The default value the function will revert to.
```
Comment (K1-R1) [inline sway-lib-std/src/option.sw:187]:
```suggestion
    /// * `default`: [T] - The default value the function will revert to.
```

**Review by K1-R1 — COMMENTED** [review]
Very nice work, just one typo mistake and then happy to approve

**Review by K1-R1 — APPROVED** [review]

**Comment by bitzoic** [comment]
This will need to be updated for https://github.com/FuelLabs/sway-standards/pull/12

**Review by K1-R1 — DISMISSED** [review]
LGTM

**Comment by Braqzen** [comment]
Currently reviewing, 31 files to go

**Review by Braqzen — COMMENTED** [review]
I haven't checked the updates to read/writes. Probably missed some stuff since there's a lot to look over. Low-key want to vomit and feel unwell now

reason_to_smile_joke

**Review by Braqzen — COMMENTED** [review]
I haven't checked the updates to read/writes. Probably missed some stuff since there's a lot to look over. Low-key want to vomit and feel unwell now

reason_to_smile_joke

File: sway-lib-std/src/call_frames.sw [hunk 2]
```diff
@@ -20,13 +20,49 @@ const SECOND_PARAMETER_OFFSET: u64 = 74;
...
+/// # Examples
+///
+/// ```sway
+/// use std::{call_frames::contract_id, token::mint};
+///
+/// fn foo() {
+///     let this_contract = contract_id();
+///     mint(50);
+///     // Transfer the newly minted tokens to the ZERO_B256 address from this contract
+///     Address::from(ZERO_B256).transfer(50, this_contract)
```
[the remark this answers is not in the document]
Comment (bitzoic) [reply sway-lib-std/src/call_frames.sw:42]: This is sending the 50 tokens to the `ZERO_B256` address

File: sway-lib-std/src/constants.sw [hunk 3]
```diff
@@ -4,7 +4,35 @@ library;
 use ::contract_id::ContractId;
 
 /// The `BASE_ASSET_ID` represents the base asset of a chain.
-/// This is currently hard coded as a zero address, but will be configurable in the future.
+///
+/// # Additional Information
+///
+/// On the Fuel network, the base asset is Ether. It is hardcoded as the 0x00..00 ContractId.
```
[the remark this answers is not in the document]
Comment (bitzoic) [reply sway-lib-std/src/constants.sw:10]: https://twitter.com/IAmNickDodson/status/1547973492770869248
https://twitter.com/IAmNickDodson/status/1548777678660042752

[the remark this answers is not in the document]
Comment (Braqzen) [reply sway-lib-std/src/constants.sw:10]: yh, we talked about this internally

File: sway-lib-std/src/call_frames.sw (hunk 2 above)
[the remark this answers is not in the document]
Comment (SwayStar123) [reply sway-lib-std/src/call_frames.sw:42]: you are right about the signature, but .transfer transfers to the address. like in solidity. It mints to the contract and then transfers from the contract to address

[the remark this answers is not in the document]
Comment (Braqzen) [reply sway-lib-std/src/call_frames.sw:42]: I see. I'm reading the specs on [tro](https://specs.fuel.network/master/fuel-vm/instruction-set.html#tro-transfer-coins-to-output) and I think I have a better idea now. I think we may be able to refactor the usage of `output_type` and `output_amount` in that function. I don't understand why the `output_amount` is 0 though. It seems like it should panic if that's the case

[the remark this answers is not in the document]
Comment (SwayStar123) [reply sway-lib-std/src/call_frames.sw:42]: I already answered this on slack once. The tro opcode panics when its 0. As for why the code is written that why you will have to blame the original transfer function in token.sw and ask whoever wrote that. This function is very similar to that one

**Review by Braqzen — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

File: sway-lib-std/src/hash.sw
Comment (Braqzen) [inline sway-lib-std/src/hash.sw:1]: Geez, I've just looked at that Hasher above. That is an atrocity. Why does it even exist?

**Review by Braqzen — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Review by sdankel — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]
Nice work!

**Pull request merged**